docs/core/utilities.md

```diff
@@ -0,0 +1,256 @@
+  hashType: 'data1',
+});
+```
+
+### isSporeScriptSupported
+
+Returns a boolean indicating weather if the target ScriptId exists in the SporeConfig.
+If "scriptName" is passed, it also checks whether the name of the target script matches.
+
+```typescript
+declare function isSporeScriptSupported(config: SporeConfig, scriptId: ScriptId, scriptName?: string): boolean;
+```
+
+#### Example
+
+Check if the target cell is any script supported by the SporeConfig:
+
+```typescript
+const cell: Cell = { ... };
+const isSupported = isSporeScriptSupported(config, cell.cellOutput.type, 'Spore');
+console.log(isSupported); // true or false
+```
+
+Check if the target cell is a Spore:
+
+```typescript
+const cell: Cell = { ... };
+const isSupported = isSporeScriptSupported(config, cell.cellOutput.type, 'Spore');
+console.log(isSupported); // true or false
+```
+
+## RawString
+
+### bytifyRawString
+
+Pack UTF-8 raw string to Uint8Array.
+
+```typescript
+declare function bytifyRawString(text: string): Uint8Array;
+```
+
+#### Example
+
+```typescript
+const bytes: Uint8Array = bytifyRawString('hello world');
+console.log(bytes.hexify(bytes)); // 0x07000000636f6e74656e74
+```
+
+### bufferToRawString
+
+Unpack Uint8Array to UTF-8 raw string.
+
+```typescript
+declare function bufferToRawString(source: BytesLike, options?: TextDecodeOptions): string;
+```
+
+#### Example
+
+```typescript
+const text: string = bufferToRawString('0x07000000636f6e74656e74');
+console.log(text); // content
+```
+
+## ContentType
+
+### encodeContentType
+
+Convert ContentType object to string.
+
+```typescript
+declare function encodeContentType(encodable: EncodableContentType): string;
+```
+
+#### Example
+
+```typescript
+const contentType = encodeContentType({
+  type: 'image',
+  subtype: 'svg+xml',
+  parameters: {
+    immortal: true,
+    q: 0.9,
+  },
+});
+console.log(contentType); // 'image/svg+xml;immortal=true;q=0.9'
+```
+
+### decodeContentType
+
+Convert string to ContentType object.
+
+```typescript
+declare function decodeContentType(contentType: string): DecodedContentType;
+```
+
+#### Example
+
+```typescript
+const decoded = decodeContentType('image/svg+xml;immortal=true;q=0.9');
+console.log(decoded); // { type: 'image', subtype: 'svg+xml', mediaType: 'image/svg+xml', parameters: { immortal: true, q: 0.9 } }
+```
+
+### isContentTypeValid
+
+Check if the target ContentType's format is valid.
+
+```typescript
+declare function isContentTypeValid(contentType: string): boolean;
+```
+
+#### Example
+
+```typescript
+const result1 = isContentTypeValid('image/svg+xml;immortal=true;q=0.9');
+console.log(result1); // true
+
+const result2 = isContentTypeValid('image/;test=1');
+console.log(result2); // false
+```
+
+## Transaction
+
+### waitForTransaction
+
+Wait for a transaction to be committed on-chain.
+Will throw an error if the transaction is not committed within some times.
```

Comment:
How long would it take before it throws an error?

Reply:
The wait time is dynamic for each turn, I'll say approximately 2 mins.